fuse_models/src/odometry_2d_publisher.cpp

```diff
@@ -245,63 +248,137 @@ void Odometry2DPublisher::tfPublishTimerCallback(const ros::TimerEvent& event)
+      covariance(0, 0) = odom_output_.pose.covariance[0];
+      covariance(0, 1) = odom_output_.pose.covariance[1];
+      covariance(0, 2) = odom_output_.pose.covariance[5];
+      covariance(1, 0) = odom_output_.pose.covariance[6];
+      covariance(1, 1) = odom_output_.pose.covariance[7];
+      covariance(1, 2) = odom_output_.pose.covariance[11];
+      covariance(2, 0) = odom_output_.pose.covariance[30];
+      covariance(2, 1) = odom_output_.pose.covariance[31];
+      covariance(2, 2) = odom_output_.pose.covariance[35];
+
+      covariance(3, 3) = odom_output_.twist.covariance[0];
+      covariance(3, 4) = odom_output_.twist.covariance[1];
+      covariance(3, 5) = odom_output_.twist.covariance[5];
+      covariance(4, 3) = odom_output_.twist.covariance[6];
+      covariance(4, 4) = odom_output_.twist.covariance[7];
+      covariance(4, 5) = odom_output_.twist.covariance[11];
+      covariance(5, 3) = odom_output_.twist.covariance[30];
+      covariance(5, 4) = odom_output_.twist.covariance[31];
+      covariance(5, 5) = odom_output_.twist.covariance[35];
+
+      // TODO(efernandez) for now we set the top right and bottom left corners to zero, but we could cache them in
+      // another attribute when we retrieve the covariance from the ceres problem
+      covariance.topRightCorner<3, 3>().setZero();
+      covariance.bottomLeftCorner<3, 3>().setZero();
+
+      const Eigen::Matrix<double, 6, 6> J = jacobian.topLeftCorner<6, 6>();
+      covariance = J * covariance * J.transpose();
+      covariance.noalias() += dt * params_.process_noise_covariance.topLeftCorner<6, 6>();
+
+      odom_output_.pose.covariance[0] = covariance(0, 0);
+      odom_output_.pose.covariance[1] = covariance(0, 1);
+      odom_output_.pose.covariance[5] = covariance(0, 2);
+      odom_output_.pose.covariance[6] = covariance(1, 0);
+      odom_output_.pose.covariance[7] = covariance(1, 1);
+      odom_output_.pose.covariance[11] = covariance(1, 2);
+      odom_output_.pose.covariance[30] = covariance(2, 0);
+      odom_output_.pose.covariance[31] = covariance(2, 1);
+      odom_output_.pose.covariance[35] = covariance(2, 2);
+
+      odom_output_.twist.covariance[0] = covariance(3, 3);
+      odom_output_.twist.covariance[1] = covariance(3, 4);
+      odom_output_.twist.covariance[5] = covariance(3, 5);
+      odom_output_.twist.covariance[6] = covariance(4, 3);
+      odom_output_.twist.covariance[7] = covariance(4, 4);
+      odom_output_.twist.covariance[11] = covariance(4, 5);
+      odom_output_.twist.covariance[30] = covariance(5, 3);
+      odom_output_.twist.covariance[31] = covariance(5, 4);
+      odom_output_.twist.covariance[35] = covariance(5, 5);
     }
-    catch (const std::exception& e)
-    {
-      ROS_WARN_STREAM_THROTTLE(5.0, "Could not lookup the " << params_.base_link_frame_id << "->" <<
-        params_.odom_frame_id << " transform. Error: " << e.what());
 
-      return;
-    }
+    odom_pub_.publish(odom_output_);
```

Comment:
@svwilliams @ayrton04 I just fixed this. I have this instruction inside the `if (params_.predict_to_current_time)` block, so if that parameter was `false`, the odometry was never published. Now it's always published, either with the optimization or predicted value. Let me know if you think something is wrong with this.

Reply:
I wrote the comment before pushing the changes, but you can simply search for `publish(` and se e that now is outside that `if` block.